### PR TITLE
NIFI-7030 Add Kerberos principal and password properties to Solr proc…

### PR DESCRIPTION
**Pull request by bbende**
…essors.

This allows authenticating to a kerberized Solr using a principal and password, as an alternative to the keytab credential service. 

I tested this with docker-kdc, Solr 8.4.1, and ZooKeeper 3.4.6, all running locally.

The Solr reference guide has good instructions on how to configure everything:

https://lucene.apache.org/solr/guide/8_4/kerberos-authentication-plugin.html#kerberos-authentication-plugin

### CODE REVIEW

File: nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java [hunk 1]
```diff
@@ -149,6 +151,25 @@
             .required(false)
             .build();
 
+    public static final PropertyDescriptor KERBEROS_PRINCIPAL = new PropertyDescriptor.Builder()
+            .name("Kerberos Principal")
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java]: Minor issue but I would use `.name("kerberos-principal")` and `.displayName("Kerberos Principal")` here.

File: nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java [hunk 2]
```diff
@@ -149,6 +151,25 @@
...
+            .description("The principal to use when specifying the principal and password directly in the processor for authenticating to Solr via Kerberos.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(AttributeExpression.ResultType.STRING))
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
+            .build();
+
+    public static final PropertyDescriptor KERBEROS_PASSWORD = new PropertyDescriptor.Builder()
+            .name("Kerberos Password")
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java]: Minor issue but I would use `.name("kerberos-password")` and `.displayName("Kerberos Password")` here.

File: nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java [hunk 3]
```diff
@@ -149,6 +151,25 @@
...
+            .description("The password to use when specifying the principal and password directly in the processor for authenticating to Solr via Kerberos.")
+            .required(false)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .addValidator(StandardValidators.createAttributeExpressionLanguageValidator(AttributeExpression.ResultType.STRING))
+            .expressionLanguageSupported(ExpressionLanguageScope.VARIABLE_REGISTRY)
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/main/java/org/apache/nifi/processors/solr/SolrUtils.java]: I don't think sensitive property descriptors should support EL - parameters now solve the requirement. This also means that in other locations where the value is referenced from the context, we do not need to evaluate EL.

File: nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/test/java/org/apache/nifi/processors/solr/TestPutSolrContentStream.java [hunk 4]
```diff
@@ -459,6 +460,67 @@ public void testBasicAuthAndKerberosNotAllowedTogether() throws IOException, Ini
...
+        runner.setProperty(SolrUtils.KERBEROS_PASSWORD, kerberosPassword);
+
+        runner.assertNotValid();
+
+        runner.removeProperty(SolrUtils.BASIC_USERNAME);
+        runner.removeProperty(SolrUtils.BASIC_PASSWORD);
+        runner.assertValid();
+
+        proc.onScheduled(runner.getProcessContext());
+        final KerberosUser kerberosUser = proc.getMockKerberosKeytabUser();;
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/test/java/org/apache/nifi/processors/solr/TestPutSolrContentStream.java]: Double `;` typo.

File: nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/test/java/org/apache/nifi/processors/solr/TestPutSolrContentStream.java [hunk 5]
```diff
@@ -459,6 +460,67 @@ public void testBasicAuthAndKerberosNotAllowedTogether() throws IOException, Ini
...
+        runner.setProperty(SolrUtils.KERBEROS_CREDENTIALS_SERVICE, "kerb-credentials");
+
+        runner.assertNotValid();
+
+        runner.removeProperty(SolrUtils.KERBEROS_PRINCIPAL);
+        runner.removeProperty(SolrUtils.KERBEROS_PASSWORD);
+        runner.assertValid();
+
+        proc.onScheduled(runner.getProcessContext());
+        final KerberosUser kerberosUser = proc.getMockKerberosKeytabUser();;
```
Comment (alopresto) [inline nifi-nar-bundles/nifi-solr-bundle/nifi-solr-processors/src/test/java/org/apache/nifi/processors/solr/TestPutSolrContentStream.java]: Double `;` typo.

**Comment by bbende** [comment]
Pushed commits that address review comments from @alopresto

**Pull request closed without merging**